cmd/producer/main.go

```diff
@@ -1,17 +1,22 @@
 package main
 
 import (
+	"context"
+	"time"
+
+	kafka "github.com/ONSdigital/dp-kafka"
 	"github.com/ONSdigital/dp-observation-importer/event"
 	"github.com/ONSdigital/dp-observation-importer/schema"
-	"github.com/ONSdigital/go-ns/kafka"
-	"time"
 )
 
 func main() {
+	ctx := context.Background()
 	var brokers []string
 	brokers = append(brokers, "localhost:9092")
 
-	producer, _ := kafka.NewProducer(brokers, "observation-extracted", int(2000000))
+	pChannels := kafka.CreateProducerChannels()
+
+	producer, _ := kafka.NewProducer(ctx, brokers, "observation-extracted", int(2000000), pChannels)
```

Comment:
Should we be assigning and dealing with the potential error here?

Reply:
Probably but not sure when this application ever gets used, it is not defined in the Makefile and probably should just be ripped out altogether.

I have just implemented the same logic as it existed before but switched out the kafka library

This is scope creep from the story I was working on, but I can handle the error just so it is tidier

Reply:
So I've read Mat Ryers [blog](https://pace.dev/blog/2020/02/12/why-you-shouldnt-use-func-main-in-golang-by-mat-ryer) and the main reason to create a run function (in my opinion) is to write tests which I do agree with. Writing these tests would increase the scope of this work and was not previously planned; so Im sorry but Im not going to do anymore on this.

There needs to be stories in the backlog to fix this and do it properly including tests.